File: matrixmulown.java
import java.util.Random;

public class MatrixMultiplicationExample {
    public static void main(String[] args) {
        int n = 10; // size of the matrices
        int numMatrices = 10000; // number of matrices to multiply
        double[][][] matrices = new double[numMatrices][n][n]; // an array to hold the matrices
        
        // fill the matrices with random values
        Random rand = new Random();
        for (int i = 0; i < numMatrices; i++) {
            for (int j = 0; j < n; j++) {
                for (int k = 0; k < n; k++) {
                    matrices[i][j][k] = rand.nextDouble();
                }
            }
        }
        
        // multiply the matrices
        double[][] result = matrices[0];
        long startTime = System.currentTimeMillis(); // start timing the execution
        for (int i = 1; i < numMatrices; i++) {
            result = multiply(result, matrices[i]);
        }
        long endTime = System.currentTimeMillis(); // end timing the execution and print the duration
        System.out.println("Execution time: " + (endTime - startTime) + " milliseconds");
    }
    
    // helper function to multiply two matrices
    public static double[][] multiply(double[][] a, double[][] b) {
        int n = a.length;
        int m = b[0].length;
        int p = b.length;
        double[][] result = new double[n][m];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                for (int k = 0; k < p; k++) {
                    result[i][j] += a[i][k] * b[k][j];
                }
            }
        }
        return result;
    }
}

File: matrixmulown.py
import random
import time

# Function to perform matrix multiplication
def matrix_multiply(mat1, mat2):
    rows1 = len(mat1)
    cols1 = len(mat1[0])
    rows2 = len(mat2)
    cols2 = len(mat2[0])

    # Check if the matrices can be multiplied
    if cols1 != rows2:
        print("Error: Number of columns in the first matrix must be equal to the number of rows in the second matrix")
        return None

    # Create a result matrix filled with zeros
    result = [[0 for _ in range(cols2)] for _ in range(rows1)]

    # Perform matrix multiplication using nested loops
    for i in range(rows1):
        for j in range(cols2):
            for k in range(rows2):
                result[i][j] += mat1[i][k] * mat2[k][j]

    return result

# Create 30,000 random 10x10 matrices
matrices = []
for _ in range(10000):
    matrix = [[random.randint(1, 10) for _ in range(10)] for _ in range(10)]
    matrices.append(matrix)

# Start the timer
start_time = time.time()

# Multiply the matrices sequentially
result = matrices[0]
for i in range(1, len(matrices)):
    result = matrix_multiply(result, matrices[i])

# Stop the timer
end_time = time.time()

# Calculate the execution time
execution_time = end_time - start_time

# Print the result and execution time
print("Result:")
print(result)
print("Execution time: {:.6f} seconds".format(execution_time))
